test/e2e/nbthread_test.go

```diff
@@ -0,0 +1,128 @@
+		t.Fatalf("failed to observe expected conditions: %v", err)
+	}
+
+	if err := nbthreadWaitForRouterDeployment(t, kclient, nbthreadRouterDeployTimeout, ic, 7); err != nil {
+		t.Fatalf("expected router deployment to set spec.threading.count: %v", err)
+	}
+
+	// set threading.count to 0, as if it was unset
+	if err := nbthreadSetThreadCount(t, kclient, 1*time.Minute, 0, ic); err != nil {
+		t.Fatalf("failed to update ingresscontroller: %v", err)
+	}
+
+	// when set back to default, the router deployment should have ROUTER_THREADS set to 4
+	if err := nbthreadWaitForRouterDeployment(t, kclient, nbthreadRouterDeployTimeout, ic, 4); err != nil {
+		t.Fatalf("expected router deployment to have spec.threading.count unset: %v", err)
+	}
+}
+
+func nbthreadGetIngressController(t *testing.T, client client.Client, timeout time.Duration) (*operatorv1.IngressController, error) {
+	ic := operatorv1.IngressController{}
+	if err := wait.PollImmediate(1*time.Second, timeout, func() (bool, error) {
+		if err := client.Get(context.TODO(), defaultName, &ic); err != nil {
+			t.Logf("Get %q failed: %v, retrying...", defaultName, err)
+			return false, nil
+		}
+		return true, nil
+	}); err != nil {
+		return nil, fmt.Errorf("failed to get %q: %v", defaultName, err)
+	}
+	return &ic, nil
+}
+
+func nbthreadWaitForRouterDeployment(t *testing.T, client client.Client, timeout time.Duration, c *operatorv1.IngressController, nbthreads int) error {
+	return wait.PollImmediate(1*time.Second, timeout, func() (bool, error) {
+		deployment := &appsv1.Deployment{}
+		if err := client.Get(context.TODO(), controller.RouterDeploymentName(c), deployment); err != nil {
+			t.Logf("Get %q failed: %v, retrying...", controller.RouterDeploymentName(c), err)
+			return false, nil
+		}
+		return nbthreadVerifyRouterDeployment(deployment, nbthreads), nil
+	})
+}
+
+func nbthreadSetThreadCount(t *testing.T, client client.Client, timeout time.Duration, nbthreads int, c *operatorv1.IngressController) error {
+	name := types.NamespacedName{Namespace: c.Namespace, Name: c.Name}
+
+	return wait.PollImmediate(1*time.Second, timeout, func() (bool, error) {
+		if err := client.Get(context.TODO(), name, c); err != nil {
+			t.Logf("Get %q failed: %v, retrying...", name, err)
+			return false, nil
+		}
+		c.Spec.Threading.Count = nbthreads
+		if err := client.Update(context.TODO(), c); err != nil {
+			t.Logf("Update %q failed: %v, retrying...", name, err)
+			return false, nil
+		}
+		return true, nil
+	})
+}
+
+func nbthreadVerifyRouterDeployment(deployment *appsv1.Deployment, nbthreads int) bool {
+	expectedThreads := nbthreads
+	if expectedThreads == 0 {
+		expectedThreads = 4
+	}
+	foundThreads := nbthreadGetThreadsFromEnv(deployment.Spec.Template.Spec.Containers[0].Env)
+	return foundThreads == expectedThreads
+}
+
+func nbthreadGetThreadsFromEnv(env []corev1.EnvVar) int {
```

Comment:
Would `waitForDeploymentEnvVar` work in place of these new functions?

https://github.com/openshift/cluster-ingress-operator/blob/master/test/e2e/operator_test.go#L1864

Reply:
That is a good fit, thanks!